### PR TITLE
Unit test rollup

### PR DESCRIPTION
**Pull request by paul-szczepanek-arm**
ready

### CODE REVIEW

File: src/controller/tests/TestWriteChunking.cpp [hunk 1]
```diff
@@ -232,10 +250,8 @@ void TestWriteChunking::TestListChunking(nlTestSuite * apSuite, void * apContext
         // Since bugs can happen, we don't want this test to never stop, so create a ceiling for how many
         // times this can run without seeing expected results.
         //
-        for (int j = 0; j < 10 && writeCallback.mOnDoneCount == 0; j++)
-        {
-            ctx.DrainAndServiceIO();
-        }
+
+        DrainAndServiceIOUntilDone(ctx, writeCallback.mSuccessCount);
```
Comment (ATmobica) [inline src/controller/tests/TestWriteChunking.cpp:254]: Do we have the same behavior as before? They added some limitations for running attempts.
The same bellow.

File: src/controller/tests/data_model/TestRead.cpp [hunk 2]
```diff
@@ -227,6 +227,17 @@ bool IsClusterDataVersionEqual(const app::ConcreteClusterPath & aConcreteCluster
...
+{
+    int32_t lastNum;
+    // keep processing while there are responses coming in
+    do
+    {
+        lastNum = *processedNum;
+        ctx.DrainAndServiceIO();
+    } while (lastNum != *processedNum);
+}
+
```
Comment (ATmobica) [inline src/controller/tests/data_model/TestRead.cpp:240]: I wonder if it wouldn't be better to put this function in a separate common file and include it in all tests that use it.

File: src/controller/tests/data_model/TestRead.cpp [hunk 3]
```diff
@@ -1033,6 +1047,7 @@ void TestReadInteraction::TestReadSubscribeAttributeResponseWithCache(nlTestSuit
         NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
 
         ctx.DrainAndServiceIO();
```
Comment (ATmobica) [inline src/controller/tests/data_model/TestRead.cpp:1049]: Shouldn't be changed to `DrainAndServiceIOUntilDone`?

File: src/controller/tests/data_model/TestRead.cpp [hunk 4]
```diff
@@ -2952,7 +2969,8 @@ void TestReadInteraction::TestReadHandler_KillOverQuotaSubscriptions(nlTestSuite
                        2);
 
     app::InteractionModelEngine::GetInstance()->ShutdownActiveReads();
-    ctx.DrainAndServiceIO();
+    ctx.GetIOContext().DriveIOUntil(System::Clock::Seconds16(60),
+                                    [&]() { return app::InteractionModelEngine::GetInstance()->GetNumDirtySubscriptions() == 0; });
```
Comment (ATmobica) [inline src/controller/tests/data_model/TestRead.cpp:2973]: Why do we add this?

File: src/inet/tests/TestInetCommonPosix.cpp [hunk 5]
```diff
@@ -461,8 +461,13 @@ void ServiceEvents(uint32_t aSleepTimeMilliseconds)
...
                 [](intptr_t) -> void {
-                    // this will allow the RunEventLoop to return
-                    chip::DeviceLayer::PlatformMgr().StopEventLoopTask();
+                    // this will allow us to drain the queue including events added after starting the queue but not forever
+                    chip::DeviceLayer::PlatformMgr().ScheduleWork(
+                        [](intptr_t) -> void {
+                            // this will allow the RunEventLoop to return
+                            chip::DeviceLayer::PlatformMgr().StopEventLoopTask();
+                        },
+                        (intptr_t) nullptr);
```
Comment (ATmobica) [inline src/inet/tests/TestInetCommonPosix.cpp:470]: Do we still need this after changes in event loop?

File: src/lib/support/tests/TestCHIPArgParser.cpp [hunk 6]
```diff
@@ -779,7 +779,12 @@ int TestCHIPArgParser(void)
     UnknownOptionTest_UnknownShortOptionBeforeKnown();
     UnknownOptionTest_UnknownLongOptionAfterArgs();
     UnknownOptionTest_IgnoreUnknownShortOption();
+#if CHIP_CONFIG_NON_POSIX_LONG_OPT
+    /* skip this test because the parser succesfully captures all the options
+       but the error reporting is incorrect in this case due to long_opt limitations */
+#else
     UnknownOptionTest_IgnoreUnknownLongOption();
+#endif // CHIP_CONFIG_NON_POSIX_LONG_OPT
```
Comment (ATmobica) [inline src/lib/support/tests/TestCHIPArgParser.cpp:787]:
```suggestion
    /* skip this test because the parser successfully captures all the options
       but the error reporting is incorrect in this case due to long_opt limitations */
#ifndef CHIP_CONFIG_NON_POSIX_LONG_OPT
    UnknownOptionTest_IgnoreUnknownLongOption();
#endif // !CHIP_CONFIG_NON_POSIX_LONG_OPT
```

File: src/platform/openiotsdk/PlatformManagerImpl.cpp [hunk 7]
```diff
@@ -146,6 +143,13 @@ void PlatformManagerImpl::RunEventLoopInternal()
             flags = osEventFlagsGet(mPlatformFlags);
         }
 
+        if (flags & kTaskStopEventFlag)
+        {
+            osEventFlagsClear(mPlatformFlags, kTaskStopEventFlag);
+            osEventFlagsClear(mPlatformFlags, kTaskRunningEventFlag);
+            break;
+        }
+
```
Comment (ATmobica) [inline src/platform/openiotsdk/PlatformManagerImpl.cpp:152]: Should all posted events be handled before stopping the event loop? Now, I think terminated event loop has a higher priority.

File: src/platform/openiotsdk/SystemTimeSupport.cpp [hunk 8]
```diff
@@ -69,30 +70,36 @@ uint64_t GetClock_MonotonicMS(void)
...
     aCurTime = Clock::Milliseconds64(GetClock_MonotonicMS());
-    return CHIP_NO_ERROR;
+    return rtcValid ? CHIP_NO_ERROR : CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
 }
 
 CHIP_ERROR ClockImpl::SetClock_RealTime(Clock::Microseconds64 aNewCurTime)
 {
     uint64_t newTickCount;
+    // if we manually set the RTC it means we can provide it now (instead of fake one)
+    rtcValid     = true;
```
Comment (ATmobica) [inline src/platform/openiotsdk/SystemTimeSupport.cpp:86]: I think we shouldn't support the RTC time at all. Both getter and setter should return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE. The InitClock_RealTime() is not required.

File: src/transport/raw/tests/TestPeerAddress.cpp [hunk 9]
```diff
@@ -93,8 +93,11 @@ void TestToString(nlTestSuite * inSuite, void * inContext)
     {
         IPAddress::FromString("1223::3456:789a", ip);
         PeerAddress::UDP(ip, 8080).ToString(buff);
+        // IPV6 does not specify case
```
Comment (ATmobica) [inline src/transport/raw/tests/TestPeerAddress.cpp:96]:
```suggestion
        // LwIP IPV6 address conversion does not specify the case
```

File: src/inet/tests/TestInetCommonPosix.cpp (hunk 5 above)
Comment (paul-szczepanek-arm) [reply src/inet/tests/TestInetCommonPosix.cpp:470]: yes

File: src/controller/tests/TestEventChunking.cpp [hunk 10]
```diff
@@ -287,14 +287,26 @@ void GenerateEvents(nlTestSuite * apSuite, chip::EventNumber & firstEventNumber,
...
+
 /*
  * This validates all the various corner cases encountered during chunking by
  * artificially reducing the size of a packet buffer used to encode attribute & event data
  * to force chunking to happen over multiple packets even with a small number of attributes or events
  * and then slowly increasing the available size by 1 byte in each test iteration and re-running
  * the report generation logic. This 1-byte incremental approach sweeps through from a base scenario of
  * N attributes fitting in a report, to eventually resulting in N+1 attributes or events fitting in a report.
-
+ *
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply src/controller/tests/TestEventChunking.cpp:309]: no

File: src/lib/support/tests/TestCHIPArgParser.cpp (hunk 6 above)
Comment (paul-szczepanek-arm) [reply src/lib/support/tests/TestCHIPArgParser.cpp:787]: I can't work out what the change here is

File: src/platform/openiotsdk/PlatformManagerImpl.cpp (hunk 7 above)
Comment (paul-szczepanek-arm) [reply src/platform/openiotsdk/PlatformManagerImpl.cpp:152]: No, the loop needs to exit in case events are generated faster than they are dispatched, hence exit comes first

File: src/platform/openiotsdk/SystemTimeSupport.cpp (hunk 8 above)
Comment (paul-szczepanek-arm) [reply src/platform/openiotsdk/SystemTimeSupport.cpp:86]: I see no harm, others support it, if the application sets the clock it will report correct time, if it's not set then it will return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE

File: src/transport/raw/tests/TestPeerAddress.cpp (hunk 9 above)
Comment (paul-szczepanek-arm) [reply src/transport/raw/tests/TestPeerAddress.cpp:96]: No, it's not LwIP that does not specify case, IP6 is a not designed by lwip. It's the IP6 specification that doe not specify case, the 'LwIP IPV6 address conversion' most certainly does and that's the problem

File: src/lib/support/tests/TestCHIPArgParser.cpp (hunk 6 above)
Comment (ATmobica) [reply src/lib/support/tests/TestCHIPArgParser.cpp:787]: Only cosmetic, but for me, it is more readable to use #ifndef instead of #if ... #else

File: src/platform/openiotsdk/SystemTimeSupport.cpp (hunk 8 above)
Comment (ATmobica) [reply src/platform/openiotsdk/SystemTimeSupport.cpp:86]: I'm not sure if we provide "real" time in this case. I think it will be safer to return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE now. And if Open IoT SDK supports RTC add it here. CYW30739 or  webos platform do this way. And `#if 0` in `InitClock_RealTime()` doesn't look good

File: scripts/examples/openiotsdk_example.sh [hunk 11]
```diff
@@ -30,14 +30,23 @@ Options:
...
+    -p,--path       <build_path>    Build path <build_path - default is example_dir/build>
 
 Examples:
     shell
     unit-tests
+
+You can run individual test suites of unit tests by using their names with the run command:
+
 EOF
+    cat /workspaces/connectedhomeip/src/test_driver/openiotsdk/unit-tests/testnames.txt
```
Comment (ATmobica) [inline scripts/examples/openiotsdk_example.sh:42]: Use CHIP_ROOT here or relative path here. The absolute path can be different for each instance.

File: scripts/examples/openiotsdk_example.sh [hunk 12]
```diff
@@ -30,14 +30,23 @@ Options:
...
+    echo ""
 }
 
+readarray -t TEST_NAMES </workspaces/connectedhomeip/src/test_driver/openiotsdk/unit-tests/testnames.txt
+
+export FAST_MODEL_PLUGINS_PATH=/workspaces/connectedhomeip
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply scripts/examples/openiotsdk_example.sh:48]: thanks for catching that, that is useful only to me

File: .vscode/tasks.json [hunk 13]
```diff
@@ -364,8 +364,34 @@
...
+                "ChipCryptoTests",
+                "MinimalMdnsRespondersTests",
+                "DataModelTests",
+                "RetransmitTests",
+                "ControllerTests",
+                "SecureChannelTests",
+                "BDXTests",
+                "PlatformTests",
+                "CredentialsTest",
+                "accesstest"
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply .vscode/tasks.json]: added

File: src/controller/tests/TestReadChunking.cpp [hunk 14]
```diff
@@ -43,10 +43,11 @@
 #include <nlunit-test.h>
 #include <utility>
 
-using TestContext = chip::Test::AppContext;
 using namespace chip;
 using namespace chip::app::Clusters;
 
+using TestContext = chip::Test::AppContext;
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply src/controller/tests/TestReadChunking.cpp:49]: @hugolhst

File: src/controller/tests/TestReadChunking.cpp [hunk 15]
```diff
@@ -213,16 +232,17 @@ class TestMutableAttrAccess
...
     }
 
     // These setters
     void SetVal(uint8_t attribute, uint8_t newVal)
     {
-        uint8_t index = static_cast<uint8_t>(attribute - 1);
-        if (index < ArraySize(val) && val[index] != newVal)
+        uint8_t index   = static_cast<uint8_t>(attribute - 1);
+        uint8_t arrsize = ArraySize(val);
+        if (index < arrsize && val[index] != newVal)
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply src/controller/tests/TestReadChunking.cpp:245]: @hugolhst

File: src/controller/tests/data_model/TestRead.cpp (hunk 2 above)
Comment (paul-szczepanek-arm) [reply src/controller/tests/data_model/TestRead.cpp:240]: Yeah, we ended using is in a couple of places, I'll try and find some location for it

File: src/platform/openiotsdk/SystemTimeSupport.cpp (hunk 8 above)
Comment (paul-szczepanek-arm) [reply src/platform/openiotsdk/SystemTimeSupport.cpp:86]: OK, I'll remove all the code and return error

File: src/platform/openiotsdk/PlatformManagerImpl.cpp [hunk 16]
```diff
@@ -109,17 +109,14 @@ CHIP_ERROR PlatformManagerImpl::_PostEvent(const ChipDeviceEvent * eventPtr)
 
 void PlatformManagerImpl::HandlePostEvent()
 {
-    /* handle all events */
-    while (osMessageQueueGetCount(mQueue))
+    /* handle an event */
```
Comment (pan-) [inline src/platform/openiotsdk/PlatformManagerImpl.cpp:109]: Something is off. The event object (not the pointer) must be copied in a buffer when posted and therefore released when it is retrieved from the queue. This could be the cause of some of the issues you observed with the event queue.

File: src/controller/tests/TestWriteChunking.cpp [hunk 17]
```diff
@@ -163,20 +163,38 @@ CHIP_ERROR TestAttrAccess::Write(const app::ConcreteDataAttributePath & aPath, a
...
 
+/** Keeps calling ctx.DrainAndServiceIO() until the processedNum variable stops changing. */
+static void DrainAndServiceIOUntilDone(TestContext & ctx, volatile uint32_t & processedNum)
+{
+    uint32_t lastNum;
+    // keep processing while there are responses coming in
+    do
+    {
+        lastNum = processedNum;
+        ctx.DrainAndServiceIO();
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply src/controller/tests/TestWriteChunking.cpp:194]: Yes, but there is no API that returns the number of items left or processed.

File: src/controller/tests/TestWriteChunking.cpp [hunk 18]
```diff
@@ -163,20 +163,38 @@ CHIP_ERROR TestAttrAccess::Write(const app::ConcreteDataAttributePath & aPath, a
...
+        else
+            ChipLogError(Zcl, "Decode result: %s", err.AsString());
         return err;
     }
 
     return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
 }
 
+/** Keeps calling ctx.DrainAndServiceIO() until the processedNum variable stops changing. */
+static void DrainAndServiceIOUntilDone(TestContext & ctx, volatile uint32_t & processedNum)
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply src/controller/tests/TestWriteChunking.cpp:187]: working on factorisation, volatile is because number is incremented elsewhere, I don't care about reading the correct value as long as it stops changing

File: src/controller/tests/TestWriteChunking.cpp [hunk 19]
```diff
@@ -163,20 +163,38 @@ CHIP_ERROR TestAttrAccess::Write(const app::ConcreteDataAttributePath & aPath, a
...
     }
     if (aPath.mListOp == app::ConcreteDataAttributePath::ListOperation::AppendItem)
     {
         ByteSpan listItem;
         CHIP_ERROR err = aDecoder.Decode(listItem);
-        ChipLogError(Zcl, "Decode result: %s", err.AsString());
+        if (err == CHIP_NO_ERROR)
+            ChipLogDetail(Zcl, "Decode result: %s", err.AsString());
+        else
+            ChipLogError(Zcl, "Decode result: %s", err.AsString());
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply src/controller/tests/TestWriteChunking.cpp:179]: I'm guessing this isn't our code - they don't use braces which gives it away but yeah, let's add them

File: src/controller/tests/TestEventChunking.cpp [hunk 20]
```diff
@@ -544,10 +557,10 @@ nlTestSuite sSuite =
 
 } // namespace
 
-int TestReadChunkingTests()
+int TestEventChunkingTests()
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply src/controller/tests/TestEventChunking.cpp:561]: because it's a mistake, look at filename

File: src/controller/tests/TestEventChunking.cpp [hunk 21]
```diff
@@ -287,14 +287,26 @@ void GenerateEvents(nlTestSuite * apSuite, chip::EventNumber & firstEventNumber,
 
 } // namespace
 
+/** Keeps calling ctx.DrainAndServiceIO() until the processedNum variable stops changing. */
+static void DrainAndServiceIOUntilDone(TestContext & ctx, volatile uint32_t & processedNum)
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply src/controller/tests/TestEventChunking.cpp:292]: see comment below

File: src/access/AccessControl.cpp [hunk 22]
```diff
@@ -598,9 +598,9 @@ bool AccessControl::IsValid(const Entry & entry)
     return true;
 
 exit:
-#if CHIP_CONFIG_TEST == 0
+#if CONFIG_BUILD_FOR_HOST_UNIT_TEST == 0
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply src/access/AccessControl.cpp:601]: there is no issue, it's just CONFIG_BUILD_FOR_HOST_UNIT_TEST is an existing one, so we use that isntead of adding a new one

File: examples/platform/openiotsdk/cmake/chip.cmake [hunk 23]
```diff
@@ -24,5 +24,10 @@ set(CONFIG_CHIP_PROJECT_CONFIG "main/include/CHIPProjectConfig.h" CACHE STRING "
 set(CONFIG_CHIP_LIB_TESTS NO CACHE BOOL "")
 set(CONFIG_CHIP_LIB_SHELL NO CACHE BOOL "")
 
+set(CONFIG_CHIP_DETAIL_LOGGING YES CACHE BOOL "")
+set(CONFIG_CHIP_PROGRESS_LOGGING YES CACHE BOOL "")
+set(CONFIG_CHIP_AUTOMATION_LOGGING YES CACHE BOOL "")
+set(CONFIG_CHIP_ERROR_LOGGING YES CACHE BOOL "")
+
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply examples/platform/openiotsdk/cmake/chip.cmake:31]: will do, thanks

File: src/controller/tests/TestWriteChunking.cpp (hunk 18 above)
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply src/controller/tests/TestWriteChunking.cpp:188]: yeah, I wasn't sure about what c++ does with references so I stuck a volatile there to force memory access but as Vincent explained to be c++ works the same as c with a pointer. I'll remove the volatile and concurrent access is irrelevant.

File: src/platform/openiotsdk/PlatformManagerImpl.cpp (hunk 16 above)
Comment (paul-szczepanek-arm) [reply src/platform/openiotsdk/PlatformManagerImpl.cpp:109]: queue contains whole events events

**Comment by paul-szczepanek-arm** [comment]
All commits above SQUASH ME will be squashed

File: src/controller/tests/TestReadChunking.cpp [hunk 24]
```diff
@@ -322,7 +342,7 @@ void TestMutableReadCallback::OnAttributeData(const app::ConcreteDataAttributePa
                                               const app::StatusIB & aStatus)
 {
     VerifyOrReturn(apData != nullptr);
-    NL_TEST_ASSERT(gSuite, aPath.mClusterId == TestCluster::Id);
+    NL_TEST_ASSERT(gSuite, aPath.mClusterId == readTestClusterId);
 
     mAttributeCount++;
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply src/controller/tests/TestReadChunking.cpp:335]: which line are you talking about? You mean the code we haven't changed?

File: src/controller/tests/TestReadChunking.cpp [hunk 25]
```diff
@@ -824,10 +842,8 @@ void TestCommandInteraction::TestSetDirtyBetweenChunks(nlTestSuite * apSuite, vo
                                  .preworks       = { WriteAttrOp(AttrOnEp5<Attr1>, 4), WriteAttrOp(AttrOnEp5<Attr2>, 4) },
                                  .expectedValues = { { AttrOnEp5<Attr1>, 4 }, { AttrOnEp5<Attr2>, 4 }, { AttrOnEp5<Attr3>, 4 } },
                                  .attributesWithSameDataVersion = { { AttrOnEp5<Attr1>, AttrOnEp5<Attr2>, AttrOnEp5<Attr3> } } });
-
                 ChipLogProgress(DataManagement,
                                 "Case 3-2: Set dirty during chunked report by wildcard path -- new dirty attribute.");
-                app::InteractionModelEngine::GetInstance()->GetReportingEngine().SetMaxAttributesPerChunk(1);
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply src/controller/tests/TestReadChunking.cpp:830]: @hugolhst

File: src/controller/tests/TestWriteChunking.cpp (hunk 1 above)
Comment (paul-szczepanek-arm) [reply src/controller/tests/TestWriteChunking.cpp]: this is now handled by the lambda in a lambda stop

**Comment by paul-szczepanek-arm** [comment]
12 commits! there were ~40 originally

File: src/controller/tests/data_model/TestRead.cpp (hunk 3 above)
Comment (paul-szczepanek-arm) [reply src/controller/tests/data_model/TestRead.cpp:1049]: not anymore

File: src/controller/tests/TestReadChunking.cpp [hunk 26]
```diff
@@ -880,20 +896,20 @@ void TestCommandInteraction::TestSetDirtyBetweenChunks(nlTestSuite * apSuite, vo
             // The attribute failed to catch last report will be picked by this report.
             ChipLogProgress(DataManagement, "Case 1-2: Check for attributes missed last report.");
             DoTest(&readCallback, { .chunksize = 1, .expectedValues = { { AttrOnEp5<Attr3>, 4 } } });
+
+            // Destroying the read client will terminate the subscription transaction.
+            DrainAndServiceIOUntilDone(ctx, readCallback.mAttributeCount);
         }
     }
 
     chip::test_utils::SleepMillis(secondsToMilliseconds(3));
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply src/controller/tests/TestReadChunking.cpp:894]: probably not, @hugolhst could you try removing it?

File: src/controller/tests/data_model/TestRead.cpp (hunk 4 above)
Comment (hugolhst) [reply src/controller/tests/data_model/TestRead.cpp:2973]: All of this test uses DriveIOUntil instead of DrainAndServiceIO, for consistancy I think it should use it here too.

**Comment by paul-szczepanek-arm** [comment]
Waiting on CI

File: src/controller/tests/TestReadChunking.cpp (hunk 25 above)
[the remark this answers is not in the document]
Comment (ATmobica) [reply src/controller/tests/TestReadChunking.cpp:830]: If you are sure that isn't necessary it's okay with me.

File: examples/platform/openiotsdk/cmake/chip.cmake [hunk 27]
```diff
@@ -24,5 +24,12 @@ set(CONFIG_CHIP_PROJECT_CONFIG "main/include/CHIPProjectConfig.h" CACHE STRING "
 set(CONFIG_CHIP_LIB_TESTS NO CACHE BOOL "")
 set(CONFIG_CHIP_LIB_SHELL NO CACHE BOOL "")
 
+set(CONFIG_CHIP_DETAIL_LOGGING YES CACHE BOOL "Enable logging at detail level")
+set(CONFIG_CHIP_PROGRESS_LOGGING YES CACHE BOOL "Enable logging at progress level")
+set(CONFIG_CHIP_AUTOMATION_LOGGING YES CACHE BOOL "Enable logging at automation level")
+set(CONFIG_CHIP_ERROR_LOGGING YES CACHE BOOL "Enable logging at error level")
+
+list(APPEND CHIP_CFLAGS \"-Wno-unused-function\")
```
Comment (ATmobica) [inline examples/platform/openiotsdk/cmake/chip.cmake:32]: Can we move this to `config/openiotsdk/CMakeLists.txt`?

File: src/controller/tests/TestEventChunking.cpp [hunk 28]
```diff
@@ -422,6 +423,7 @@ void TestReadEvents::TestMixedEventsAndAttributesChunking(nlTestSuite * apSuite,
 
         NL_TEST_ASSERT(apSuite, readClient.SendRequest(readParams) == CHIP_NO_ERROR);
 
+        ctx.DrainAndServiceIO();
```
Comment (pan-) [inline src/controller/tests/TestEventChunking.cpp:426]: Why do we need it twice ?

**Review by pan- — COMMENTED** [review]
I'm having some issues running the tests locally, it ends with: 

```
[ Test-CHIP-CryptoContext ]
[ERR] [TST] Tests already running
```

**Comment by pan-** [comment]
On my system, it seems that I receive several network up events: 

Not exiting when it is the case seems to solve it. See: https://github.com/ATmobica/connectedhomeip/blob/development/src/test_driver/openiotsdk/unit-tests/main/main.cpp#L51.

File: .github/workflows/examples-openiotsdk.yaml [hunk 29]
```diff
@@ -79,8 +79,9 @@ jobs:
                     examples/shell/openiotsdk/build/chip-openiotsdk-shell-example.elf \
                     /tmp/bloat_reports/
 
-            - name: Build unit tests
+            - name: Build and run unit tests
               if: github.event_name == 'push' || steps.changed_paths.outputs.openiotsdk == 'true'
-              timeout-minutes: 20
+              timeout-minutes: 100
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply .github/workflows/examples-openiotsdk.yaml:84]: this steps runs unit tests, takes half an hour but not sure how long in CI

[the remark this answers is not in the document]
Comment (ATmobica) [reply .github/workflows/examples-openiotsdk.yaml]: Ok, so maybe we can set 40 minutes here? Should be good enough.

File: src/controller/tests/TestEventChunking.cpp (hunk 28 above)
Comment (paul-szczepanek-arm) [reply src/controller/tests/TestEventChunking.cpp:426]: thanks for catching this, must be accidental paste when I was removing the serviceuntil change

**Comment by paul-szczepanek-arm** [comment]
> On my system, it seems that I receive several network up events:
> 
> Not exiting when it is the case seems to solve it. See: https://github.com/ATmobica/connectedhomeip/blob/development/src/test_driver/openiotsdk/unit-tests/main/main.cpp#L51.

Thanks, same happened in CI.

File: .github/workflows/examples-openiotsdk.yaml (hunk 29 above)
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply .github/workflows/examples-openiotsdk.yaml]: 60 OK?

**Review by ATmobica — APPROVED** [review]

**Review by pan- — APPROVED** [review]
